### ISSUE TITLE
object should have etag when gateway is started with --etag.

### ISSUE THREAD

**Issue by zhoucheng361** [juicedata/juicefs#5010]
**What happened**:
```
root@bench-01:~/juicefs# ./juicefs format sqlite3://test.db test
2024/07/19 16:05:03.596102 juicefs[2148417] <INFO>: Meta address: sqlite3://test.db [interface.go:504]
2024/07/19 16:05:03.596689 juicefs[2148417] <INFO>: Data use file:///var/jfs/test/ [format.go:484]
2024/07/19 16:05:03.663798 juicefs[2148417] <INFO>: Volume is formatted as {
  "Name": "test",
  "UUID": "05bdf5ac-c267-4703-b8d6-d4cdc88c10f2",
  "Storage": "file",
  "Bucket": "/var/jfs/",
  "BlockSize": 4096,
  "Compression": "none",
  "EncryptAlgo": "aes256gcm-rsa",
  "TrashDays": 1,
  "MetaVersion": 1,
  "MinClientVersion": "1.1.0-A",
  "DirStats": true,
  "EnableACL": false
} [format.go:521]
root@bench-01:~/juicefs# MINIO_ROOT_USER=minioadmin MINIO_ROOT_PASSWORD=minioadmin ./juicefs gateway  sqlite3://test.db localhost:9006 --keep-etag -d --multi-buckets
2024/07/19 16:06:41.292124 juicefs[2148505] <INFO>: Meta address: sqlite3://test.db [interface.go:504]
2024/07/19 16:06:41.292617 juicefs[2148505] <INFO>: open log file /var/log/juicefs-gateway.log: %!s(<nil>) [mount_unix.go:248]
2024/07/19 16:06:41.793750 juicefs[2148505] <INFO>: OK, service is ready on localhost:9006 [mount_unix.go:222]
root@bench-01:~/juicefs# mc alias set myminio http://localhost:9006 minioadmin minioadmin
Added `myminio` successfully.
root@bench-01:~/jfs# mc mb myminio/abc
Bucket created successfully `myminio/abc`.
root@bench-01:~/jfs# mc cp README.md myminio/abc/
README.md:         1.40 KiB / 1.40 KiB ┃▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓┃ 64.66 KiB/s 0sroot@bench-01:~/jfs# mc ls myminio/abc/README.md
[2024-07-19 20:41:28 CST] 1.4KiB README.md
root@bench-01:~/jfs# mc ls myminio/abc/README.md  --json
{
 "status": "success",
 "type": "file",
 "lastModified": "2024-07-19T20:41:28.653+08:00",
 "size": 1435,
 "key": "README.md",
 "etag": "",
 "url": "http://localhost:9006/abc/README.md",
 "versionOrdinal": 1
}
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

**Comment by zhijian-pro** [comment]
<img width="920" alt="image" src="https://github.com/user-attachments/assets/52345506-6d92-4b33-a34e-42ecb58350a5">


<img width="1162" alt="image" src="https://github.com/user-attachments/assets/9addca34-9799-4f7d-adc5-36ca8b79cd31">

**Comment by zhoucheng361** [comment]
```
root@bench-01:~/jfs# mc mb myminio/abc
Bucket created successfully `myminio/abc`.
root@bench-01:~/jfs# mc cp README.md myminio/abc/
README.md:         1.40 KiB / 1.40 KiB ┃▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓┃ 64.66 KiB/s 0sroot@bench-01:~/jfs# mc ls myminio/abc/README.md
[2024-07-19 20:41:28 CST] 1.4KiB README.md
root@bench-01:~/jfs# mc ls myminio/abc/README.md  --json
{
 "status": "success",
 "type": "file",
 "lastModified": "2024-07-19T20:41:28.653+08:00",
 "size": 1435,
 "key": "README.md",
 "etag": "",
 "url": "http://localhost:9006/abc/README.md",
 "versionOrdinal": 1
}
```

**Comment by zhoucheng361** [comment]
fixed by #5013
